### ISSUE TITLE
Generic metadata in .any.js tests

### ISSUE THREAD

**Issue by muodov** [web-platform-tests/wpt#26285]
At the moment, .any.js tests support just a few `// META: ` entries such as `title`, `timeout`, and a few specific entries such as `script`. Other metadata entries seem to be silently ignored.

Window tests widely use `<meta>` tags such as "author" and "help", and it's currently not clear from the docs whether it's safe to use `// META: author=John Doe`.

So my questions are:
- what should be the official way of specifying generic metadata in non-HTML tests?
- what are the caveats of just allowing any type of `// META: foo=bar`?
- HTML meta tag can contain more than just a name/content, e.g. `<meta charset="...">`. Would it be useful to support this structure in .any.js?

**Comment by annevk** [comment]
FWIW, I don't mind removing this metadata. I don't think recording authorship was ever a good idea and the help stuff is routinely out-of-date and I'm not aware of people actively using that in their workflow.

**Comment by muodov** [comment]
Are there other more important tags? I guess "help" links to the relevant spec sections might be worth keeping?

**Comment by annevk** [comment]
Maybe, but historically at least that often got out-of-date. Perhaps things are more stable these days,  but I'd still be a bit wary.

**Comment by stephenmcgruer** [comment]
cc @web-platform-tests/wpt-core-team since this seems to be heading towards the meta question of what metadata we want to support (pun intended).

**Comment by Ms2ger** [comment]
Do we not lint against invalid keys? We might need to document that.

Help links can be in plain comments - that also allows matching them to subtests or specific assertions, rather than just the entire file. Git already does better at authorship information. I definitely don't think either of those are compelling use cases to complicate the META feature.

**Comment by muodov** [comment]
Makes sense. Should we discourage no-op meta tags in html tests then?

**Comment by jgraham** [comment]
I agree in principle, but I also note that we lint CSS tests for `<link rel=help>`, so it kind of makes sense to support that one. Otherwise we should lint for things that have no effect since they could be errors.
